app/models/air_import.rb

```diff
@@ -0,0 +1,78 @@
+require 'securerandom'
+
+class AirImport < MeasurementImport
+  DATA_MESSAGE = "dat"
+
+  belongs_to :user
+
+  has_many :air_logs
+
+  store :status_details, accessors: [ :measurements_added ]
+
+  def import_measurements
+    connection = ActiveRecord::Base.connection.raw_connection
+    name = SecureRandom.uuid
+    connection.prepare name, <<-SQL
```

Comment:
This can be done with the `upsert` gem which creates a merge function automatically for you.